contracts/cw20-base/src/integration_test.rs

```diff
@@ -0,0 +1,703 @@
+      })
+    );
+
+    assert_err(res, crate::error::ContractError::CannotExceedCap {});
+  }
+
+  #[test]
+  fn should_burn_token() {
+    let (mut app, token_addr) = load_prefix();
+    let minter = get_uaddr("minter");
+    let recipient = get_uaddr("recipient");
+    let burn_amount = Uint128::from(100_000u128);
+
+    let res = exec(
+      &mut app,
+      &minter,
+      &token_addr,
+      &(crate::msg::ExecuteMsg::Mint {
+        recipient: recipient.to_string(),
+        amount: Uint128::from(1_000_000u128),
+      })
+    );
+
+    assert!(res.is_ok());
+
+    let total_supply_before_burn: cw20::TokenInfoResponse = query(&app, &token_addr, &(crate::msg::QueryMsg::TokenInfo {})).unwrap();
+
+    let res = exec(
+      &mut app,
+      &recipient,
+      &token_addr,
+      &(crate::msg::ExecuteMsg::Burn {
+        amount: burn_amount,
+      })
+    );
+
+    assert!(res.is_ok());
+
+    // Check recipient balance
+    let recipient_balance: cw20::BalanceResponse = query(
+      &app,
+      &token_addr,
+      &(crate::msg::QueryMsg::Balance {
+        address: recipient.to_string(),
+      })
+    ).unwrap();
+
+    assert_eq!(recipient_balance.balance, total_supply_before_burn.total_supply - burn_amount);
+
+    // Check total supply
+    let token_info: cw20::TokenInfoResponse = query(&app, &token_addr, &(crate::msg::QueryMsg::TokenInfo {})).unwrap();
+    assert_eq!(token_info.total_supply, total_supply_before_burn.total_supply - burn_amount);
+  }
+
+  #[test]
+  fn should_burn_from_token() {
+    let (mut app, token_addr) = load_prefix();
+    let minter = get_uaddr("minter");
+    let recipient = get_uaddr("recipient");
+    let burner = get_uaddr("burner");
+    let burn_amount = Uint128::from(100_000u128);
+
+    let res = exec(
+      &mut app,
+      &minter,
+      &token_addr,
+      &(crate::msg::ExecuteMsg::Mint {
+        recipient: recipient.to_string(),
+        amount: Uint128::from(1_000_000u128),
+      })
+    );
+
+    assert!(res.is_ok());
+
+    let total_supply_before_burn: cw20::TokenInfoResponse = query(&app, &token_addr, &(crate::msg::QueryMsg::TokenInfo {})).unwrap();
+
+    // Approve burner
+    exec(
+      &mut app,
+      &recipient,
+      &token_addr,
+      &(crate::msg::ExecuteMsg::IncreaseAllowance {
+        spender: burner.to_string(),
+        amount: burn_amount,
+        expires: None,
+      })
+    ).unwrap();
+
+    // Burn from
+    exec(
+      &mut app,
+      &burner,
+      &token_addr,
+      &(crate::msg::ExecuteMsg::BurnFrom {
+        owner: recipient.to_string(),
+        amount: burn_amount,
+      })
+    ).unwrap();
+
+    assert!(res.is_ok());
+
+    // Check recipient balance
+    let recipient_balance: cw20::BalanceResponse = query(
+      &app,
+      &token_addr,
+      &(crate::msg::QueryMsg::Balance {
+        address: recipient.to_string(),
+      })
+    ).unwrap();
+
+    assert_eq!(recipient_balance.balance, total_supply_before_burn.total_supply - burn_amount);
```

Comment:
To clearly the assertion.
```suggestion
    let recipient_balance_expected = total_supply_before_burn.total_supply - burn_amount;
    assert_eq!(recipient_balance.balance, recipient_balance_expected);
```